streamer/src/recvmmsg.rs

```diff
@@ -44,21 +45,29 @@ pub fn recv_mmsg(socket: &UdpSocket, packets: &mut [Packet]) -> io::Result</*num
 
 #[cfg(target_os = "linux")]
 #[allow(deprecated)]
-fn cast_socket_addr(addr: &sockaddr_storage, hdr: &mmsghdr) -> Option<InetAddr> {
+fn cast_socket_addr(addr: &sockaddr_storage, hdr: &mmsghdr) -> Option<SocketAddr> {
     use libc::{sa_family_t, sockaddr_in, sockaddr_in6};
     const SOCKADDR_IN_SIZE: usize = std::mem::size_of::<sockaddr_in>();
     const SOCKADDR_IN6_SIZE: usize = std::mem::size_of::<sockaddr_in6>();
     if addr.ss_family == AF_INET as sa_family_t
         && hdr.msg_hdr.msg_namelen == SOCKADDR_IN_SIZE as socklen_t
     {
-        let addr = addr as *const _ as *const sockaddr_in;
-        return Some(unsafe { InetAddr::V4(*addr) });
+        let addr = unsafe { &*(addr as *const _ as *const sockaddr_in) };
+        return Some(SocketAddr::V4(SocketAddrV4::new(
+            Ipv4Addr::from(addr.sin_addr.s_addr.to_ne_bytes()),
+            u16::from_be(addr.sin_port),
+        )));
```

Comment:
I literally copy from socket2: https://github.com/rust-lang/socket2/blob/65085d9dff270e588c0fbdd7217ec0b392b05ef2/src/sockaddr.rs#L165-L193

Reply:
Can you please put that link as a comment in the code?

I also see they have a different code for `#[cfg(windows)]`. any reason to not include that here?

Reply:
> Can you please put that link as a comment in the code?

sure! fixed: https://github.com/anza-xyz/agave/pull/558/commits/0d04bf1c92b1ff491855f02a7d78ca895b7508a6

> I also see they have a different code for #[cfg(windows)]. any reason to not include that here?

afaik cast_socket_addr has `#[cfg(target_os = "linux")]` so it will never touch the windows condition
https://github.com/anza-xyz/agave/pull/558/files#diff-2d88fdbf6278e2d0250658ffc2903c9a6701e9b68e3e2303a23691b50bc298e2R46-R47